internal/agent/client/podman.go

```diff
@@ -0,0 +1,224 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/flightctl/flightctl/pkg/executer"
+	"github.com/flightctl/flightctl/pkg/log"
+)
+
+const (
+	podmanCmd = "podman"
+)
+
+type Podman struct {
+	exec    executer.Executer
+	log     *log.PrefixLogger
+	timeout time.Duration
+}
+
+type ImageConfig struct {
+	Labels map[string]string `json:"Labels"`
+}
+
+func NewPodman(log *log.PrefixLogger, exec executer.Executer) *Podman {
+	return &Podman{
+		log:     log,
+		exec:    exec,
+		timeout: defaultPodmanTimeout,
+	}
+}
+
+// Pull pulls the image from the registry.
+func (p *Podman) Pull(ctx context.Context, image string) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
```

Comment:
Same question about timeouts.  We may have some slow connections at the edge.

Reply:
we prefetch images where possible and all transient errors should be retired for clear signal.